software/deploy/deploy_test.go

```diff
@@ -99,6 +103,32 @@ func (s *Suite) TestValidRuntimeImageRepo() {
 	s.False(validRuntimeImageRepo("personal-repo/ap-airflow"))
 }
 
+func (s *Suite) SetupSuite() {
+	// Common setup logic for the test suite
+	s.fsForLocalConfig = afero.NewMemMapFs()
+	afero.WriteFile(s.fsForLocalConfig, config.HomeConfigFile, testUtil.NewTestConfig("localhost"), 0o777)
+
+	s.fsForDockerConfig = afero.NewMemMapFs()
+	afero.WriteFile(s.fsForLocalConfig, config.HomeConfigFile, testUtil.NewTestConfig("docker"), 0o777)
+}
+
+func (s *Suite) SetupTest() {
+	s.mockImageHandler = new(mocks.ImageHandler)
+	imageHandlerInit = func(image string) airflow.ImageHandler {
+		return s.mockImageHandler
+	}
+	s.houstonMock = new(houston_mocks.ClientInterface)
+}
+
+func (s *Suite) TearDownSuite() {
+	// Cleanup logic, if any (e.g., clearing mocks)
```

Comment:
we should assert that expectations have been met here for the mocks so that they are not written/maintained in each test